### PR TITLE
Robotized/Roundstart borgs now transfer DNA and quirks from their old body correctly

### LINKED ISSUE #9900: Cyborg cloned as human with IPC name
Issue reported from  Round ID: 45988 (BeeStation Sage - Roleplay)

Reporting client version: 514.1589

Strange thing happened but here what follows:

Selected IPC character(With robotic name, obviously)
Choosed Cyborg
Later that round, got obliterated by Ash Drake(Accidentally provoked)
Die as MMI in Lavaland due to ash storm(First time seeing it, or was on purpose due to brain decay)
Reactivated by doctor(Still being MMI)
Failed attempt of inserting MMI into cyborg shell(They said that brain is dead or too damaged)
Doctor and roboticist found out that brain inside the MMI was organic human brain, not positronic IPC brain!
Failed attempt to repair the brain with mannitol(Examining the brain said that brain is lifeless)
Initiated cloning attempt... success!
Came out as human with IPC name(And with no gender, as upon examination it says they, not he/she)

I say that it might have been overlooked with some bit of late join IPC cyborg.

### PR DESCRIPTION
**Pull request by tontyGH**
## About The Pull Request

Closes #9900

* the ticker's `equip_characters()` proc no longer checks if the character is a `mob/living/carbon/human/` at the quirk step
  * This means that the non-human jobs (AI, cyborg) will still have quirk preferences assigned
  * They will continue to not handle their quirks, however, so this shouldn't affect gameplay
* rewrites `robotize()` to use the MMI's `transfer_identity()`
* rewrites the MMI's `transfer_identity()` proc to use the `brain_item`'s `transfer_identity()`
* removed some redundant (hopefully) lines of code

## Why It's Good For The Game

this is expected behaviour

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/39193182/09342232-6b38-4d84-9b1f-7efa911fd521

</details>

## Changelog
:cl:
tweak: spawning characters will now have their quirks assigned regardless of what mob type they are
fix: roundstart cyborgs should now keep the DNA and quirks of their character
/:cl:

### CODE REVIEW

File: code/modules/mob/living/brain/MMI.dm [hunk 1]
```diff
@@ -127,25 +127,30 @@
...
-
 	if(ishuman(L))
 		var/mob/living/carbon/human/H = L
 		var/obj/item/organ/brain/newbrain = H.getorgan(/obj/item/organ/brain)
-		newbrain.forceMove(src)
-		brain = newbrain
-	else if(!brain)
+		if(newbrain)
+			. = TRUE
+			newbrain.Remove(H, TRUE) //this calls newbrain.transfer_identity()
```
Comment (Penwin0) [inline code/modules/mob/living/brain/MMI.dm:135]: Why remove and not just transfer_identity?

File: code/modules/mob/transform_procs.dm [hunk 2]
```diff
@@ -481,8 +481,6 @@
 	regenerate_icons()
 	icon = null
 	invisibility = INVISIBILITY_MAXIMUM
-	for(var/t in bodyparts)
-		qdel(t)
```
Comment (Penwin0) [inline code/modules/mob/transform_procs.dm:485]: Why?

File: code/modules/mob/living/brain/MMI.dm [hunk 3]
```diff
@@ -127,25 +127,30 @@
...
+			brain = newbrain
+
+	if(!brain)
 		brain = new(src)
-		brain.name = "[L.real_name]'s brain"
+
+	if(!brain.brainmob)
+		if(brainmob)
+			qdel(brainmob) //hopefully this isn't incredibly short sighted and ignorant and breaks everything
+		brain.transfer_identity(L)
```
Comment (Penwin0) [inline code/modules/mob/living/brain/MMI.dm:149]: Can you explain this chunk?

File: code/modules/mob/living/brain/brain_item.dm [hunk 4]
```diff
@@ -80,8 +80,6 @@
 	name = "[L.name]'s brain"
 	if(brainmob || decoy_override)
 		return
-	if(!L.mind)
-		return
```
Comment (Penwin0) [inline code/modules/mob/living/brain/brain_item.dm:84]: Does this really interfere with mmis transfering and such? Seems unrelated

File: code/modules/mob/transform_procs.dm (hunk 2 above)
Comment (tontyGH) [reply code/modules/mob/transform_procs.dm:485]: All the transformation procs qdel the mob at the end, which qdels all bodyparts anyways

It also interferes with robotize because doing this also qdeletes the brain, which we need to transfer into the cyborg's MMI

File: code/modules/mob/living/brain/MMI.dm (hunk 1 above)
Comment (tontyGH) [reply code/modules/mob/living/brain/MMI.dm:135]: `h.Remove()` removes the brain from the human's list of organs so that it doesn't get qdeleted along with the mob

File: code/modules/mob/living/brain/brain_item.dm (hunk 4 above)
Comment (tontyGH) [reply code/modules/mob/living/brain/brain_item.dm:84]: Robotize transfers the mob's mind into the cyborg before transferring identity to avoid the message that shows when you get debrained

Since mind.transfer_to() nulls the mind of the mob, it would have aborted the creation of the brainmob which we need to transfer species, DNA and stuff

File: code/modules/mob/living/brain/MMI.dm (hunk 3 above)
Comment (tontyGH) [reply code/modules/mob/living/brain/MMI.dm:149]: It's because brains don't create a brainmob when they're initialized, so we use transfer_identity() to make one

The qdel is a sanity check incase somehow the brain doesn't have a brainmob, but the MMI does, which shouldn't happen

**Review by Penwin0 — APPROVED** [review]

**Comment by tontyGH** [comment]
Oh cool

File: code/modules/mob/living/brain/MMI.dm (hunk 3 above)
Comment (Rukofamicom) [reply code/modules/mob/living/brain/MMI.dm:149]: This does not account for roundstart (and latejoin) cyborgs. 

When any of these cyborgs has their MMI removed via any means it triggers a runtime and forcibly ejects the player to a ghost permanently.
